### PR TITLE
Add default time selectors to metrics config

### PR DESCRIPTION
**Pull request by djbarnwal**
- [x] Handle cases with unsupported user inputs in the YAML
- [x] Invalidate UI state when default are changed
- [x] Integrate available time grains with the dashboard controls

### CODE REVIEW

File: web-common/src/components/floating-element/WithTogglableFloatingElement.svelte [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
   import { FloatingElement } from "./index";
   import Portal from "../Portal.svelte";
+  export let disabled = false;
```
Comment (hamilton) [inline web-common/src/components/floating-element/WithTogglableFloatingElement.svelte:7]: Can you give insight into why this was needed? It seems to me if a float element needs to be disabled, you can control that at the trigger point (usually the primary slot that uses `toggleFloatingElement`)

File: web-common/src/features/dashboards/time-controls/TimeRangeNameSelector.svelte [hunk 2]
```diff
@@ -0,0 +1,101 @@
+<script lang="ts">
```
Comment (hamilton) [inline web-common/src/features/dashboards/time-controls/TimeRangeNameSelector.svelte:1]: I think this file may be redundant and we're using `TimeRangeSelector.svelte` now.

**Review by hamilton — APPROVED** [review]
Code-wise, this looks good to me. You and I went through all the functionality so I feel confident about the parts we worked on together. It looks like a test will have to be refactored for this.

I left a question & a note about a file that I don't think we're using anymore that should be deleted from this PR. Once those are addressed, think another pass from @magorlick might be helpful. Ideally we can get this merged tomorrow.

A bit concerned by the amount of reactive statement logic here, but let's clean this up when we circle back on the metrics config.

File: web-common/src/components/floating-element/WithTogglableFloatingElement.svelte (hunk 1 above)
Comment (djbarnwal) [reply web-common/src/components/floating-element/WithTogglableFloatingElement.svelte:7]: Ah, good catch. We were using this before for the selectors but seems like it's not needed now.

**Pull request merged**